lightning-invoice/src/utils.rs

```diff
@@ -1886,4 +1891,154 @@ mod test {
 			_ => panic!(),
 		}
 	}
+
+    const TEST_RESULT_SIZE: usize = 3;
+
+    #[test]
+    fn test_zip_vecs_one() {
```

Comment:
Why add more tests? Do the current set of tests not cover the previously implemented behavior?

Reply:
I am not sure which tests you are referencing.

I added multiple tests for the different cases of inputs with expected results.  If there is a way to parameterize a single test function by defining a set of inputs with their paired expected output, I can change the test to that.

If you are referencing an existing set of tests that I overlook, let me know.

Reply:
> If you are referencing an existing set of tests that I overlook, let me know.

Yeah, I'm referring to the existing set of tests. Presumably, they would be testing the previous logic so should be sufficient to test the refactored logic. Just wondering if this is actually the case or if instead you thought the previous testing was insufficient.

Reply:
After I worked through where the `select_phantom_hints` was being called, I realized what I missed about your question.  You were asking what about the existing tests and I misunderstood it as "existing unit tests" rather than the "existing integration tests".

I think there is value in having unit tests around this function.  My reasoning is that it tests the different inputs without adding those different cases to the larger integration test.  This does two things, in my opinion, allow for devs to make improvements to the function with the unit tests as an assurance of correctness and also allow devs to see what the expected functionality is through example (tests being the example of input -> output).

If there is a convention in this project that I am new to and I should remove these unit tests, I can take them out.